### PR TITLE
Sqlite

### PR DESCRIPTION
**Pull request by xyzhu**
Please pull the sqlite branch.

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -10,51 +10,34 @@ Database:
...
-log4j-1.2.16.jar
-mysql-connector-java-5.1.13-bin.jar
-slf4j-api-1.6.1.jar
-slf4j-simple-1.6.1.jar
-joda-time-1.6.2-sources.jar
-joda-time-1.6.2.jar
+Maven:
 
+FixCache is managed by maven. So you have to install maven for tesing, running FixCache.
+Using maven, you have to put the pom.xml file in the project path.
```
Comment (cflewis) [inline README:18]: This isn't exactly true. You have to run Maven from where the pom.xml is stored. This should also say what goals are supported (I'm guessing compile, run, test?)

File: README [hunk 2]
```diff
@@ -10,51 +10,34 @@ Database:
...
-UserPass:
-
-If you want to run fixcache on a larger project, you will need to create a database.properties file for that project.
-
+URL:jdbc:mysql://localhost:3306/(your database name)
+UserName:(the user name of your database)
+UserPass:(the password of your database)
+ 
 To build fixcache:
 ./BUILD
```
Comment (cflewis) [inline README:32]: Why do we still need a BUILD script? Why isn't Maven able to do this? The only reason you'd want a build script is if you have to run some particular commands from the command-line, but I don't see what those would be here.

If Maven isn't being used for building in the script (it's not been changed in this request, so I doubt it), then the dependencies it downloads will not be made available on the classpath.

File: README [hunk 3]
```diff
@@ -10,51 +10,34 @@ Database:
...
-Once you have an empty test database and a testdatabase.properties file set up, you can run the FixCache testing suite.
-
-./TEST
-
-All tests should pass. 
-
-To run FixCache on a larger database, set up a database.properties file, check that the project id matches the one specified by the RUN script -p flag, and then use the RUN script. 
+Once you have all the data for a given project in the database and the database.properties file set up, you can run the FixCache.
 
 ./RUN
```
Comment (cflewis) [inline README:42]: Again, why isn't Maven being used here? It seems to me that the RUN script isn't being altered in this pull request, so dependencies that Maven downloads will not be exported to the class path, so it won't work.

File: TEST [hunk 4]
```diff
@@ -1,7 +1,7 @@
```
Comment (cflewis) [inline TEST:0]: Do you need this file anymore? Why isn't Maven running the tests?

**Comment by cflewis** [comment]
This pull request is not ready yet, I have concerns about whether Maven is being used correctly (see my comments). Either these Maven changes should be removed (I don't think this is a good idea) or addressed (better!).

**Comment by cflewis** [comment]
I've looked at the repo. Maven is not being used correctly at all. Maven is not some fancy way of downloading dependencies (although it does that). It's a build system, and a run system, but the clean, build, test, run scripts are all still there. A mainClass should be setup for the exec:run goal, and also for the JAR packager. You can see this in my [Lakitu pom](https://github.com/Lewisham/Lakitu/blob/master/pom.xml).

Additionally, the package definitions don't make any sense. `src/main/java/FixCache/Cache/Cache.java` should not mean the package is `Cache`. The package is `FixCache.Cache`. The whole codebase should be rearranged so that it's in a package that makes sense, something like `edu.ucsc.sil.fixcache` (note the lower case), which would be `src/main/java/edu/ucsc/sil/fixcache/`. Same goes for tests.

These fixes won't take long, but probably shouldn't be part of this pull request. If I get time, I can do the fork and do it myself, but it's probably not going to happen until Friday.

File: TEST (hunk 4 above)
Comment (xyzhu) [inline TEST:0]: The TEST script is not used now.

Comment (cflewis) [inline TEST:0]: If you don't use it, you should delete it from the repository.

**Comment by cflewis** [comment]
I've issued a pull request for you to get changes so Maven works right and the packages are laid out correctly. You should merge this in.

Additionally, your unit tests are failing on my computer (see log below) and every test throws a warning out to the command-line. This means you are not ready for merging yet: you should never merge something that doesn't pass unit tests. I know you've been having trouble getting the tests to work with Maven, but you're going to need to work out how to fix it or have a good reason why it can't be fixed.

```
java.sql.SQLException: The database file is locked
    at org.sqlite.DB.execute(DB.java:359)
    at org.sqlite.PrepStmt.executeQuery(PrepStmt.java:77)
    at Util.Database.getIntResult(Database.java:12)
    at Cache.CacheItem.findLoc(CacheItem.java:281)
    at Cache.CacheItem.update(CacheItem.java:116)
    at Cache.CacheItem.<init>(CacheItem.java:85)
    at Cache.Cache.add(Cache.java:114)
    at Cache.Simulator.initialPreLoad(Simulator.java:279)
    at Test.SimulatorTest.testSimulate(SimulatorTest.java:122)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
    at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:76)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
    at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
    at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
    at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:31)
    at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
    at org.apache.maven.surefire.junit4.JUnit4TestSet.execute(JUnit4TestSet.java:62)
    at org.apache.maven.surefire.suite.AbstractDirectoryTestSuite.executeTestSet(AbstractDirectoryTestSuite.java:140)
    at org.apache.maven.surefire.suite.AbstractDirectoryTestSuite.execute(AbstractDirectoryTestSuite.java:127)
    at org.apache.maven.surefire.Surefire.run(Surefire.java:177)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.maven.surefire.booter.SurefireBooter.runSuitesInProcess(SurefireBooter.java:345)
    at org.apache.maven.surefire.booter.SurefireBooter.main(SurefireBooter.java:1009)
```

This is the warning. You should find a way of fixing this, disabling this if you are sure it doesn't matter, or do as they request and go to the forums.

```
6757 [main] WARN org.dbunit.dataset.AbstractTableMetaData - Potential problem found: The configured data type factory 'class org.dbunit.dataset.datatype.DefaultDataTypeFactory' might cause problems with the current database 'SQLite' (e.g. some datatypes may not be supported properly). In rare cases you might see this message because the list of supported database products is incomplete (list=[derby]). If so please request a java-class update via the forums.If you are using your own IDataTypeFactory extending DefaultDataTypeFactory, ensure that you override getValidDbProducts() to specify the supported database products.
```

**Comment by cflewis** [comment]
Are you opening more than one connection to the SQLite database? That would cause the problem here.

**Comment by xyzhu** [comment]
All the test can pass on my computer. I do not know how to resolve the warning, so I just let it be there.

**Comment by cflewis** [comment]
Leaving the warning isn't acceptable, as it means we can't deploy the code anywhere else. You need to fix it.

How are you running the tests? `mvn test` spits out all those errors. Additionally, you don't need to make the temporary SQLite table, you can just use `conn = DriverManager.getConnection("jdbc:sqlite:memory");` to make an in-memory database. That means you won't be making files that aren't going to be cleaned up.

**Comment by xyzhu** [comment]
I just use "mvn test" and all of them can pass.

**Pull request merged**